inc/class-wpseo-utils.php

```diff
@@ -1135,13 +1135,62 @@ public static function is_plugin_network_active() {
 		return $network_active;
 	}
 
+	/**
+	 * Gets the type of the current post.
+	 */
+	public static function get_post_type() {
+		global $post;
+
+		if ( isset( $post->post_type ) ) {
+			return $post->post_type;
+		}
+		elseif ( ! isset( $post->post_type ) && isset( $_GET['post_type'] ) ) {
+			return sanitize_text_field( $_GET['post_type'] );
+		}
+
+		return '';
+	}
+
+	/**
+	 * Gets the type of the current page.
+	 */
+	public static function get_page_type() {
+		global $pagenow;
+		if ( WPSEO_Metabox::is_post_edit( $pagenow ) ) {
+			return 'post';
+		}
+
+		return 'taxonomy';
+	}
+
 	/**
 	 * Getter for the Adminl10n array. Applies the wpseo_admin_l10n filter.
 	 *
 	 * @return array The Adminl10n array.
 	 */
 	public static function get_admin_l10n() {
-		$wpseo_admin_l10n = [];
+		$taxonomy_labels = WPSEO_Taxonomy::get_labels();
+
+		$post_type = WPSEO_Utils::get_post_type();
+		$page_type = WPSEO_Utils::get_page_type();
+
+		/* Adjust the no-index text strings based on the post type. */
+		$post_type_object = get_post_type_object( $post_type );
+
+		$label_array = [
```

Comment:
It doesn't seem that this array is used for anything except to grab values out of it for `$wpseo_admin_l10n`. Define these values in the array below.